Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -443,21 +433,12 @@ def _set_id_channel(channel_id):
         self.boosting_manager.add_source(dispersy_cid)
         chn_obj = self.boosting_manager.get_source_object(dispersy_cid)
 
-        def _load(torrent, callback=None):
-            if not isinstance(torrent, CollectedTorrent):
-                torrent_id = 0
-                if torrent.torrent_id <= 0:
-                    torrent_id = self.session.lm.torrent_db.getTorrentID(torrent.infohash)
-                if torrent_id:
-                    torrent.update_torrent_id(torrent_id)
-
-                torrent = CollectedTorrent(torrent, self.tdef)
-            if callback is not None:
-                callback(torrent)
-            else:
-                return torrent
+        def _load(_):
+            defer_ret = defer.Deferred()
+            defer_ret.callback(self.tdef)
+            return defer_ret
```

Comment:
Same comment as above. Also, can this method be moved to a super class (since you are using it in two places)?